include/private/soci-vector-helpers.h

```diff
@@ -0,0 +1,111 @@
+// (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+//
+
+#ifndef SOCI_VECTOR_HELPERS_H_INCLUDED
+#define SOCI_VECTOR_HELPERS_H_INCLUDED
+
+#include "soci-exchange-cast.h"
+
+namespace soci
+{
+
+namespace details
+{
+
+// Helper functions to work with vectors.
+
+// Get the size of the vector.
+inline std::size_t get_vector_size(exchange_type e, void *data)
+{
+    switch (e)
+    {
+        case x_char:
+            return exchange_vector_type_cast<x_char>(data).size();
+        case x_stdstring:
+            return exchange_vector_type_cast<x_stdstring>(data).size();
+        case x_short:
+            return exchange_vector_type_cast<x_short>(data).size();
+        case x_integer:
+            return exchange_vector_type_cast<x_integer>(data).size();
+        case x_long_long:
+            return exchange_vector_type_cast<x_long_long>(data).size();
+        case x_unsigned_long_long:
+            return exchange_vector_type_cast<x_unsigned_long_long>(data).size();
+        case x_double:
+            return exchange_vector_type_cast<x_double>(data).size();
+        case x_stdtm:
+            return exchange_vector_type_cast<x_stdtm>(data).size();
+        case x_xmltype:
+            return exchange_vector_type_cast<x_xmltype>(data).size();
+        case x_longstring:
+            return exchange_vector_type_cast<x_longstring>(data).size();
+        default:
+            throw soci_error("Failed to get the size of the vector of non-supported type.");
+    }
+}
+
+// Set the size of the vector.
+inline void resize_vector(exchange_type e, void *data, std::size_t newSize)
+{
+    switch (e)
+    {
+        case x_char:
+            exchange_vector_type_cast<x_char>(data).resize(newSize);
+            break;
+        case x_stdstring:
+            exchange_vector_type_cast<x_stdstring>(data).resize(newSize);
+            break;
+        case x_short:
+            exchange_vector_type_cast<x_short>(data).resize(newSize);
+            break;
+        case x_integer:
+            exchange_vector_type_cast<x_integer>(data).resize(newSize);
+            break;
+        case x_long_long:
+            exchange_vector_type_cast<x_long_long>(data).resize(newSize);
+            break;
+        case x_unsigned_long_long:
+            exchange_vector_type_cast<x_unsigned_long_long>(data).resize(newSize);
+            break;
+        case x_double:
+            exchange_vector_type_cast<x_double>(data).resize(newSize);
+            break;
+        case x_stdtm:
+            exchange_vector_type_cast<x_stdtm>(data).resize(newSize);
+            break;
+        case x_xmltype:
+            exchange_vector_type_cast<x_xmltype>(data).resize(newSize);
+            break;
+        case x_longstring:
+            exchange_vector_type_cast<x_longstring>(data).resize(newSize);
+            break;
+        default:
+            throw soci_error("Failed to get the size of the vector of non-supported type.");
```

Comment:
Same as above, except that here you will either use an extra var or replace `break`s above with `return`s.